crates/ruff_server/src/server/api/requests/code_action_resolve.rs

```diff
@@ -75,6 +80,43 @@ pub(super) fn resolve_edit_for_fix_all(
     Ok(action)
 }
 
+pub(super) fn resolve_edit_for_organize_imports(
+    mut action: types::CodeAction,
+    document: &crate::edit::Document,
+    url: &types::Url,
+    mut linter_settings: ruff_linter::settings::LinterSettings,
```

Comment:
Hmm, it's unfortunate that this requires making `LinterSettings` cloneable but I haven't been able to come up with an alternative solution that not at least requires cloning individual fields. Ideally it would be possible to call `check` with the desired rules, but that's a larger refactor and is probably not worth it.

I don't feel strongly but I have a slight preference to pass in a `LinterSetting` and move the cloning into `resolve_edit_for_organize_imports`.

Reply:
Could we do something like the following?
1. Take a snapshot of existing settings
2. Mutate it temporarily
3. Restore the settings with the snapshot

I'm not sure how involved this is or if it's better than cloning. That said, I wouldn't recommend doing in this PR but something to think about if we need or want to refactor this in the future.

Reply:
@MichaReiser @dhruvmanila I'll make a follow-up issue to look into how we could improve this for the future.